cookbook/bundles/best_practices.rst

```diff
@@ -72,49 +68,44 @@ class name.
     :class:`Symfony\\Bundle\\FrameworkBundle\\FrameworkBundle`.
 
 Each bundle has an alias, which is the lower-cased short version of the bundle
-name using underscores (``acme_hello`` for ``AcmeHelloBundle``, or
-``acme_social_blog`` for ``Acme\Social\BlogBundle`` for instance). This alias
-is used to enforce uniqueness within a bundle (see below for some usage
-examples).
+name using underscores (``acme_blog`` for ``AcmeBlogBundle``). This alias
+is used to enforce uniqueness within a bundle (see below for some usage examples).
 
 Directory Structure
 -------------------
 
-The basic directory structure of a HelloBundle must read as follows:
+The basic directory structure of a AcmeBlogBundle must read as follows:
 
 .. code-block:: text
 
-    XXX/...
-        HelloBundle/
-            HelloBundle.php
-            Controller/
-            Resources/
-                meta/
-                    LICENSE
-                config/
-                doc/
-                    index.rst
-                translations/
-                views/
-                public/
-            Tests/
-
-The ``XXX`` directory(ies) reflects the namespace structure of the bundle.
-
-The following files are mandatory:
-
-* ``HelloBundle.php``;
+    <your-project>/
+    ├─ ...
+    └─ src/
+       └─ Acme/
+          └─ BlogBundle/
```

Comment:
well, in any case, we don't really care about how you structure your repo, as the autoloading is handled by the composer metadata anyway. There are several ways:

- PSR-4 at the root, as done for PSR-0 with target-dir previously
- code of the bundle inside a ``src`` folder (which could allow you to have tests in a separate folder rather than putting them in the bundle source itself, but also allows to have folders for PhpSpec, Behat or Atoum for instance)

The convention of putting the bundle at the root of the repo itself is a pre-PSR-4 convention when putting it inside a ``src`` folder would force to have a deep path to the source code (which was also painful well when viewing the source on Github until a few months ago).

The best practice should refer to the structure inside the bundle only. The structure of the whole repo is up to the developer accoding to their own development preferences or to requirements of tools they rely on.